src/binlog/events/transaction_payload_event.rs

```diff
@@ -0,0 +1,296 @@
+                            let val = match buf.read_lenenc_int() {
+                                Ok(val) => val,
+                                Err(_) => {
+                                    let length = buf.eat_lenenc_int();
+                                    buf.skip(length as usize);
+                                    Err(io::Error::new(
+                                    io::ErrorKind::InvalidData,
+                                    format!("Error reading val for OTW_PAYLOAD_COMPRESSION_TYPE_FIELD"),
+                                ))?
+                                }
+                            };
+                            ob.algorithm =
+                                TransactionPayloadCompressionType::try_from(val).unwrap();
+                            have_compression_type = true;
+                            continue;
+                        }
+                        Ok(TransactionPayloadFields::OTW_PAYLOAD_UNCOMPRESSED_SIZE_FIELD) => {
+                            let _length = match buf.read_lenenc_int() {
+                                Ok(_) => {}
+                                Err(_) => {
+                                    let length = buf.eat_lenenc_int();
+                                    buf.skip(length as usize);
+                                    Err(io::Error::new(
+                                    io::ErrorKind::InvalidData,
+                                    format!("Error reading length for OTW_PAYLOAD_UNCOMPRESSED_SIZE_FIELD"),
+                                ))?
+                                }
+                            };
+                            let val = match buf.read_lenenc_int() {
+                                Ok(val) => val,
+                                Err(_) => {
+                                    let length = buf.eat_lenenc_int();
+                                    buf.skip(length as usize);
+                                    Err(io::Error::new(
+                                    io::ErrorKind::InvalidData,
+                                    format!("Error reading val for OTW_PAYLOAD_UNCOMPRESSED_SIZE_FIELD"),
+                                ))?
+                                }
+                            };
+                            ob.uncompressed_size = RawInt::new(val);
+                            continue;
+                        }
+                        Err(val) => {
+                            let length = buf.eat_lenenc_int();
+                            buf.skip(length as usize);
+                            Err(io::Error::new(
+                                io::ErrorKind::InvalidData,
+                                format!("Unknown field type: {:?}", val),
+                            ))?;
+                        }
+                    }
+                }
+                Err(val) => {
+                    let length = buf.eat_lenenc_int();
+                    buf.skip(length as usize);
```

Comment:
net_field_length format error seem to be fatal, and there is nothing to skip here.
I believe we'd better to replace this `match buf.read_lenenc_int() { .. }` with just `let val = buf.read_lenenc_int()?`.

Reply:
Done